keras/src/layers/preprocessing/image_preprocessing/random_apply.py

```diff
@@ -0,0 +1,151 @@
+        self.generator = SeedGenerator(seed)
+        self.built = True
+
+    def get_random_transformation(self, data, training=True, seed=None):
+        if not training:
+            return None
+
+        if seed is None:
+            seed = self._get_seed_generator(self.backend._backend)
+
+        if isinstance(data, dict):
+            inputs = data["images"]
+        else:
+            inputs = data
+
+        input_shape = self.backend.shape(inputs)
+        if self.batchwise:
+            should_augment = self._rate > self.backend.random.uniform(
+                shape=(), seed=seed
+            )
+        else:
+            batch_size = input_shape[0]
+            random_values = self.backend.random.uniform(
+                shape=(batch_size,), seed=seed
+            )
+            should_augment = random_values < self._rate
+
+            ndims = len(input_shape)
+            ones = [1] * (ndims - 1)
+            broadcast_shape = tuple([batch_size] + ones)
+            should_augment = self.backend.numpy.reshape(
+                should_augment, broadcast_shape
+            )
+
+        return {
+            "should_augment": should_augment,
+            "input_shape": input_shape,
+        }
+
+    def transform_images(self, images, transformation, training=True):
+        if not training or transformation is None:
+            return images
+
+        should_augment = transformation["should_augment"]
+
+        if hasattr(self._layer, "get_random_transformation"):
+            layer_transform = self._layer.get_random_transformation(
+                images, training=True
+            )
+            augmented = self._layer.transform_images(
+                images, layer_transform, training=True
+            )
+        else:
+            augmented = self._layer(images)
+
+        return self.backend.numpy.where(should_augment, augmented, images)
+
+    def call(self, inputs):
+        if isinstance(inputs, dict):
+            return {
+                key: self._call_single(input_tensor)
+                for key, input_tensor in inputs.items()
+            }
+        else:
+            return self._call_single(inputs)
+
+    def _call_single(self, inputs):
+        transformation = self.get_random_transformation(inputs, training=True)
+        return self.transform_images(inputs, transformation, training=True)
+
+    @staticmethod
+    def _num_zero_batches(images):
+        """Count number of all-zero batches in the input."""
+        num_batches = ops.shape(images)[0]
+        flattened = ops.reshape(images, (num_batches, -1))
+        any_nonzero = ops.any(ops.not_equal(flattened, 0), axis=1)
+        num_non_zero_batches = ops.sum(ops.cast(any_nonzero, dtype="int32"))
+        return num_batches - num_non_zero_batches
+
+    def transform_labels(self, labels, transformation, training=True):
+        return labels
+
+    def transform_bounding_boxes(self, bboxes, transformation, training=True):
+        return bboxes
```

Comment:
Seems incorrect -- the bboxes should be transformed by the inner layer if applied